tests/scenarios/watch-mode-test.ts

```diff
@@ -239,13 +227,153 @@ class EmberCLI {
   }
 }
 
+class File {
+  constructor(readonly label: string, readonly fullPath: string) {}
+
+  async exists(): Promise<boolean> {
+    return pathExists(this.fullPath);
+  }
+
+  async read(): Promise<string | null> {
+    try {
+      return await fs.readFile(this.fullPath, { encoding: 'utf-8' });
+    } catch (error) {
+      if (error.code === 'ENOENT') {
+        return null;
+      } else {
+        throw error;
+      }
+    }
+  }
+
+  async write(content: string): Promise<void> {
+    await fs.writeFile(this.fullPath, content, { encoding: 'utf-8' });
+  }
+
+  async delete(): Promise<void> {
+    await fs.unlink(this.fullPath);
+  }
+}
+
+class AssertFile {
+  readonly file: File;
+
+  constructor(private assert: Assert, file: File) {
+    this.file = file;
+  }
+
+  async exists(): Promise<void> {
+    this.assert.true(await this.file.exists(), `${this.file.label} exists`);
+  }
+
+  async doesNotExist(): Promise<void> {
+    this.assert.false(await this.file.exists(), `${this.file.label} does not exists`);
+  }
+
+  async hasContent(expected: string): Promise<void> {
+    let actual = await this.file.read();
+
+    if (actual === null) {
+      this.assert.ok(false, `${this.file.label} does not exists`);
+    } else {
+      this.assert.equal(actual, expected, `content of ${this.file.label}`);
+    }
+  }
+
+  async doesNotHaveContent(expected: string | RegExp): Promise<void> {
+    let actual = await this.file.read();
+
+    if (actual === null) {
+      this.assert.ok(false, `${this.file.label} does not exists`);
+    } else {
+      this.assert.notEqual(actual, expected, `content of ${this.file.label}`);
+    }
+  }
+
+  async includesContent(expected: string): Promise<void> {
+    let actual = await this.file.read();
+
+    if (actual === null) {
+      this.assert.ok(false, `${this.file.label} does not exists`);
+    } else {
+      this.assert.true(actual.includes(expected), `content of ${this.file.label}`);
+    }
+  }
+
+  async doesNotIncludeContent(expected: string): Promise<void> {
+    let actual = await this.file.read();
+
+    if (actual === null) {
+      this.assert.ok(false, `${this.file.label} does not exists`);
+    } else {
+      this.assert.false(actual.includes(expected), `content of ${this.file.label}`);
+    }
+  }
+}
+
+function d(strings: TemplateStringsArray, ...values: unknown[]): string {
```

Comment:
common-tags does all this, yeah?
https://www.npmjs.com/package/common-tags
`stripIndent`?